Source/Dafny/AST/DafnyAst.cs

```diff
@@ -3113,6 +3159,8 @@ public abstract class Declaration : INamedRegion, IAttributeBearingDeclaration {
     public IToken tok;
     public IToken BodyStartTok = Token.NoToken;
     public IToken BodyEndTok = Token.NoToken;
+    public IToken FirstDeclarationToken = Token.NoToken;
+    public IToken TokenBeforeDocstring = Token.NoToken;
```

Comment:
Oh actually, since the trivia is attached a bit differently now, the token before the docstring actually contains the docstring in the trailing trivia of the token (see TriviaTest.cs)
The splitting occurs on the first two consecutive newlines without comments, if any.

Reply:
The name `TokenBeforeDocstring` is a little confusing for me, since our `Token` class also contains the trivia, so the trivia are not separate entities between `Token`s, leading me to read this as `TokenBeforeDocStringToken`, while it's really `TokenWithTrailingDocString`.

Reply:
Very good comment and suggestion. Renamed.